### ISSUE TITLE
Fix back navigation after "Web search" on a new tab, user should go to previous tab

### ISSUE THREAD

**Issue by subin-chella** [duckduckgo/Android#1070]
I always perform web search for words from the the current tab by doing a long press. The websearch usually happens on a new tab. Once im done with the new tab, i would like to return to my initial tab by just pressing the back button. Instead, the app is closing.

**Describe the solution or the user experience you'd like**
Always close the current tab while pressing the back button. If the user is on the last tab then close the app.

**Additional context**

**Comment by summersab** [comment]
+1 for what it's worth. I've always been slightly annoyed by this but just learned to live with it over the past few years.

**Comment by cmonfortep** [comment]
Thanks for reporting that one.
We improved our tab navigation between tabs when opening a link in a new tab or in a background tab, but the "Web search" use case is something we didn't think about.

**Comment by subin-chella** [comment]
Hi,
Created a PR with the fix.
https://github.com/duckduckgo/Android/pull/1079

**Comment by cmonfortep** [comment]
Thanks @subin-chella I've just merged the PR, it will be included in our next release. 🦆

**Comment by cmonfortep** [comment]
Hi @subin-chella we needed to rollback your contribution due to some issues found in our release.

See details here: https://github.com/duckduckgo/Android/pull/1106

TLDR; 
External links were also created having a parent tab (sourceTab Id). The contribution correctly addressed the scenario of performing a "WebSearch" but it was applying the same logic to external intents. We should try to differentiate them.

Thanks.

**Comment by TianyiMu** [comment]
Hi, I created a PR with a new fix #1154. Please take a look, thanks!

**Comment by cmonfortep** [comment]
We are closing this issue as we don't have plans in the short-mid term to address it. Sorry about that.
